unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/BackpackEditorHUDController.cs

```diff
@@ -1,43 +1,210 @@
             dataStore.HUDs.avatarEditorVisible.OnChange -= SetVisibility;
             dataStore.exploreV2.configureBackpackInFullscreenMenu.OnChange -= ConfigureBackpackInFullscreenMenuChanged;
+            dataStore.exploreV2.isOpen.OnChange -= ExploreV2IsOpenChanged;
+
+            backpackEmotesSectionController.OnNewEmoteAdded -= OnNewEmoteAdded;
+            backpackEmotesSectionController.OnEmotePreviewed -= OnEmotePreviewed;
+            backpackEmotesSectionController.OnEmoteEquipped -= OnEmoteEquipped;
+            backpackEmotesSectionController.OnEmoteUnequipped -= OnEmoteUnequipped;
+            backpackEmotesSectionController.Dispose();
+
             view.Dispose();
         }
 
         public void SetVisibility(bool visible)
         {
             if (visible)
+            {
+                backpackEmotesSectionController.LoadEmotes();
+                LoadUserProfile(ownUserProfile, true);
                 view.Show();
+            }
             else
+            {
                 view.Hide();
+                view.ResetPreviewEmote();
+            }
+
+            SetRenderScale(visible);
         }
 
         private void SetVisibility(bool current, bool _) =>
             SetVisibility(current);
 
         private void ConfigureBackpackInFullscreenMenuChanged(Transform currentParentTransform, Transform previousParentTransform) =>
             view.SetAsFullScreenMenuMode(currentParentTransform);
+
+        private void ExploreV2IsOpenChanged(bool current, bool previous)
+        {
+            if (current || !avatarIsDirty)
+                return;
+
+            avatarIsDirty = false;
+            LoadUserProfile(ownUserProfile, true);
+            backpackEmotesSectionController.RestoreEmoteSlots();
+        }
+
+        private void LoadUserProfile(UserProfile userProfile) =>
+            LoadUserProfile(userProfile, false);
+
+        private void LoadUserProfile(UserProfile userProfile, bool forceLoading)
+        {
+            bool avatarEditorNotVisible = CommonScriptableObjects.rendererState.Get() && !view.isVisible;
+            bool isPlaying = !Application.isBatchMode;
+
+            if (!forceLoading && isPlaying && avatarEditorNotVisible) return;
+            if (userProfile == null) return;
+            if (userProfile.avatar == null || string.IsNullOrEmpty(userProfile.avatar.bodyShape)) return;
+
+            wearablesCatalogService.WearablesCatalog.TryGetValue(userProfile.avatar.bodyShape, out var bodyShape);
+
+            if (bodyShape == null) return;
+            if (avatarIsDirty) return;
+
+            EquipBodyShape(bodyShape);
+            EquipSkinColor(userProfile.avatar.skinColor);
+            EquipHairColor(userProfile.avatar.hairColor);
+            EquipEyesColor(userProfile.avatar.eyeColor);
+
+            model.wearables.Clear();
+
+            int wearablesCount = userProfile.avatar.wearables.Count;
+
+            if (!DataStore.i.common.isPlayerRendererLoaded.Get()) return;
```

Comment:
please inject the datastore instead of `DataStore.i`

Reply:
Ups, I missed it. Fixed: https://github.com/decentraland/unity-renderer/pull/4933/commits/2e215f87e571783ec0bf5e41bca0f3714d300209